internal/codespaces/api/api_test.go

```diff
@@ -115,6 +115,145 @@ func TestListCodespaces_unlimited(t *testing.T) {
 	}
 }
 
+func createFakeSearchReposServer(t *testing.T, searchText string, responseRepos []*Repository) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/search/repositories" {
+			t.Fatal("Incorrect path")
```

Comment:
[t.Fatal](https://pkg.go.dev/testing#T.FailNow) "must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test". Use t.Error + return here.